composer/utils/checkpoint.py

```diff
@@ -188,6 +189,24 @@ def read_metadata(self) -> Metadata:
         return super().read_metadata()
 
 
+@retry(num_attempts=5)
+def download_object_or_file(
+    object_name: str,
+    file_destination: Union[str, Path],
+    object_store: Union[ObjectStore, LoggerDestination],
+):
+    if isinstance(object_store, ObjectStore):
+        object_store.download_object(
+            object_name=object_name,
+            filename=file_destination,
+        )
+    else:
+        object_store.download_file(
+            remote_file_name=object_name,
+            destination=str(file_destination),
+        )
```

Comment:
can we just add `download_object` as a wrapper fn to LoggerDestination and forward args to `download_file`? 

@eracah thoughts?

Reply:
@mvpatel2000  I want to minimize this PR. Can we do that in a follow-up PR if we decide to do so ?

Reply:
discussed offline with @mvpatel2000 , not a blocker for this PR.